test/collections.js

```diff
@@ -123,6 +123,11 @@ describe('Collections', function () {
       expect(map2).to.have.entries([['a', 'b'], ['c', 'd']]);
     });
 
+    it('non-entry objects should throw', function () {
+      expect(new Map([1, 2, 3])).to['throw'](TypeError);
```

Comment:
It looks like, for example, Safari's native implementation does throw here - I'll confirm what the correct behavior should be, and then either patch it or fix the shim.